cobigen/cobigen-core/src/test/java/com/devonfw/cobigen/unittest/config/reader/ContextConfigurationReaderTest.java

```diff
@@ -34,14 +34,51 @@ public void testErrorOnInvalidConfiguration() throws InvalidConfigurationExcepti
   }
 
   /**
-   * Tests whether a valid configuration can be read from src/main/templates folder
+   * Tests whether not having a context.xml top level in a template set results in an
+   * {@link InvalidConfigurationException}
+   *
+   * @throws InvalidConfigurationException expected
+   */
+  @Test(expected = InvalidConfigurationException.class)
+  public void testErrorOnInvalidConfigurationNotTopLevel() throws InvalidConfigurationException {
+
+    new ContextConfigurationReader(
+        Paths.get(new File(testFileRootPath + "invalid_source_context_not_top_level").toURI()));
+  }
+
+  /**
+   * Tests whether a valid configuration can be read from src/main/templates folder (old config location) Backward
+   * Compatibility test, remove when monolithic context.xml is deprecated.
    *
    * @throws Exception test fails
    */
   @Test
   public void testContextLoadedFromRootAndSourceFolder() throws Exception {
 
-    CobiGenFactory.create(new File(testFileRootPath + "valid_source_folder").toURI());
+    CobiGenFactory.create(new File(testFileRootPath + "valid_source_folder_root").toURI());
+  }
+
+  /**
+   * Tests whether a valid configuartion can be read from src/main/templates/templateSet (new config location)
+   *
+   * @throws Exception test fails
+   */
+  @Test
+  public void testContextLoadedFromTemplateSetFolder() throws Exception {
+
+    CobiGenFactory.create(new File(testFileRootPath + "valid_source_folder_modular").toURI());
+  }
+
+  /**
+   * Tests whether a valid configuartion can be read from both root and templateSet folders (new templates with old
+   * custom templates) Backward Compatibility test, remove when monolithic context.xml is deprecated.
+   *
+   * @throws Exception test fails
+   */
+  @Test
+  public void testContextLoadedFromTemplateSetFolderAndRoot() throws Exception {
+
+    CobiGenFactory.create(new File(testFileRootPath + "valid_source_folder_modular_and_root").toURI());
```

Comment:
Should we really consider this case?
Wouldn't it be better to support either the new or the old configuration and not a mixture of both?

Reply:
Updated the test to expect a thrown exception with a message that contains the wiki link.